Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerTransmissions/GearBox.cs

```diff
@@ -360,10 +390,26 @@ public float TractiveForceN
                     else if (Gears[0].TypeGearBox == 2 || Gears[0].TypeGearBox == 3)
                     {
 
-                        if (ClutchPercent >= -20)
+                        if (GearBoxOperation == GearBoxOperation.Manual || (GearBoxOperation != GearBoxOperation.Manual && ClutchPercent >= -20))
                         {
                             float tractiveForceN = DieselEngine.DieselTorqueTab[DieselEngine.RealRPM] / DieselEngine.DieselTorqueTab.MaxY() * CurrentGear.MaxTractiveForceN;
 
+
+                            // Limit tractive force if engine is governed, ie speed cannot exceed the governed speed
+                            if (DieselEngine.RealRPM >= DieselEngine.GovenorRPM && ShaftRPM > DieselEngine.GovenorRPM)
+                            {
+                                // use decay function to decrease tractive effort if RpM exceeds governed RpM value.
+                                // y = original amount ( 1 - decay rate)^length of prediction
+
+                                var decayRpM = ShaftRPM - DieselEngine.GovenorRPM;
+                                var teDecline = Math.Pow((1.0f - 0.05f), decayRpM);
+
+                                tractiveForceN = (float)Math.Abs(CurrentGear.MaxTractiveForceN * teDecline);
+                                tractiveForceN = MathHelper.Clamp(tractiveForceN, 0.0f, CurrentGear.MaxTractiveForceN);  // Clamp tractive effort so that it doesn't go below zero
+                            }
+
+//                            Trace.TraceInformation("Geared Tractive Effort #1 - Driving - TE: {0} lbf, RpM: {1}, Torque: {2} lb-ft, Throttle%: {3}, MaxTorque: {4} lb-ft, MaxTE {5} lbf, Speed: {6} mph, Clutch {7}, Gear: {8} IsClutchOn {9} Shaft RpM {10}", tractiveForceN * 0.224809f, DieselEngine.RealRPM, DieselEngine.DieselTorqueTab[DieselEngine.RealRPM] * 0.737562f, DieselEngine.DemandedThrottlePercent, DieselEngine.DieselTorqueTab.MaxY(), CurrentGear.MaxTractiveForceN * 0.224809f, CurrentSpeedMpS * 2.23694f, ClutchPercent, Locomotive.GearBoxController.CurrentNotch, IsClutchOn, ShaftRPM);
```

Comment:
Deactivated debug code should not be in a comment, but in a preprocessor block.

For exemple:
#if GEARBOX_DEBUG
// Debug code
#endif